README.md

```diff
@@ -275,6 +281,15 @@ The artifact checksums can be found on the respective release pages. Once the
 PR is merged, [cut a new branch](#cut-the-branch). There is no automation after
 this, so you're done here.
 
+## Release docs
+To release the documentation for the release follow [docs release process](https://knative.dev/help/maintainer/docs-release-process/)
+The current process can be summarize in the following:
+1. Wait for dependent repos to release (today thats /serving and /eventing)
```

Comment:
Reasoning why I think it doesn't belong here is because that link has all the (and then some) details (how to cut a branch, etc.) and I personally found it confusing intermingled here because it's entirely different procedure currently. Client stuff that was added recently are steps that are necessary beyond the normal steps so they made sense to be in this section. WDYT?

Reply:
What's about splitting the document into two major sections:

* Releasing the code
* Releasing the documentation

And add a top-level TOC highlighting this difference. We then can also just say "release the docs after the code" which also automatically includes all dependencies. I think that makes sense because you are then free in the docs to reference any additional repos (beyond serving and eventing) and you don't have to take care to update this release document. I see the parallelization benefit (starting the docs release before all repositories has been published) as not big enough to introduce this additional maintenance dependency (updating this doc everytime you want to depend on a new repo in the docs)